devito/core/cpu.py

```diff
@@ -35,6 +35,19 @@ class CPU64NoopOperator(OperatorCore):
     Number of CIRE passes to detect and optimize away redundant sum-of-products.
     """
 
+    CIRE_MINCOST_INV = 50
+    """
+    Minimum operation count of a Dimension-invariant aliasing expression to be
+    optimized away. Dimension-invariant aliases are lifted outside of one or more
+    invariant loop(s), so they require tensor temporaries that can be potentially
+    very large (e.g., the whole domain in the case of time-invariant aliases).
+    """
+
+    CIRE_MINCOST_SOPS = 10
+    """
+    Minimum operation count of a sum-of-product aliasing expression to be optimized away.
+    """
```

Comment:
can't we move these defaults to `OperatorCore` instead, unnecessary reapeat here.

Reply:
no, they will change at some point for GPUs. And there already are a few differences, the CIRE_REPEATS_INV, for example, is 2 in the GPU one. As you may guess, I thought carefully about centralizing this, but it was growing into a big mess, with partial overrides in the individual operators, so I gave up. I prefer the **root** Operator classes (the CPU root, and the GPU root) to define them explicitly

Reply:
Ok fine with me